DataProcessing/datax-host/src/main/scala/datax/securedsetting/KeyVaultClient.scala

```diff
@@ -33,19 +33,25 @@ object KeyVaultClient {
       return Option(secretId)
 
     secretRegex.findFirstMatchIn(secretId) match {
-      case Some(secretInfo) => val vaultName = secretInfo.group(1)
-        val secretName = secretInfo.group(2)
+      case Some(secretInfo) => val secretType = secretInfo.group(1)
+        val vaultName = secretInfo.group(2)
+		val secretName = secretInfo.group(3)
 
         cache.synchronized{
           cache.get(secretId) match {
             case Some(value) => Some(value)
             case None =>
-              val secret = kvc.synchronized{
-                kvc.getSecret(s"https://$vaultName.vault.azure.net",secretName)
-              }
+              var value = ""
+			  if(secretType == "secretscope"){
+				value = dbutils.secrets.get(scope = vaultName, key = secretName)
```

Comment:
Recommend synchronizing the calls to dbutils to prevent throttling on secret access, similar to how its done for keyvault below

Reply:
Added synchronized to dbutils